src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq_vmr.c

```diff
@@ -1554,6 +1556,41 @@ static int xgq_freq_scaling(struct platform_device *pdev,
 	return ret;
 }
 
+static int set_and_verify_freqs(struct platform_device *pdev,unsigned short *target_freqs, int num_freqs, int verify)
```

Comment:
- change the existing xgq_freq_scaling to xgq_freq_scaling_impl
- add xgq_freq_scaling(... int verify)
  example
```
xgq_freq_scaling(...int verify)
{
      xgq_freq_scaling_impl()

     if (verify) {
         xgq_freq_verify() //add your new code in this function
    }
}
```